### PR TITLE
BSTOR-97: Products should be retrieved via an API

### PR DESCRIPTION
**Pull request by Elrhino**

### CODE REVIEW

**Comment by Elrhino** [comment]
Oops, I forgot the size attribute for the t-shirts.

**Comment by christiangoudreau** [comment]
Unit tests and license headers?

File: common/src/main/java/com/arcbees/beestore/common/api/ProductResource.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 ArcBees Inc.
```
Comment (olafleur) [inline common/src/main/java/com/arcbees/beestore/common/api/ProductResource.java]: 2016

File: server/src/main/java/com/arcbees/beestore/server/api/ProductResourceImpl.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2015 ArcBees Inc.
```
Comment (olafleur) [inline server/src/main/java/com/arcbees/beestore/server/api/ProductResourceImpl.java]: 2016

**Comment by olafleur** [comment]
LGTM. Thanks for the unit tests !

**Comment by Elrhino** [comment]
@olafleur I did some refactoring. I'll need another review on this when you'll have time. Thank you!

File: client/src/test/java/com/arcbees/beestore/client/application/product/ProductPresenterTest.java [hunk 3]
```diff
@@ -34,6 +38,8 @@
 import static org.mockito.Mockito.reset;
 import static org.mockito.Mockito.verify;
 
+import static com.gwtplatform.dispatch.rest.delegates.test.DelegateTestUtils.givenDelegate;
```
Comment (olafleur) [inline client/src/test/java/com/arcbees/beestore/client/application/product/ProductPresenterTest.java:41]: should be before the other static imports, I think.

File: client/src/test/java/com/arcbees/beestore/common/dto/ProductDtoTest.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    @Test
+    public void getBrand_shouldReturnAppropriateBrand() throws Exception {
+        Brand expectedBrand = Brand.JUKITO;
+        ProductDto productDto = new ProductDto.Builder()
+                .withBrand(expectedBrand)
+                .build();
+
+        assertTrue(expectedBrand.equals(productDto.getBrand()));
+    }
+}
```
Comment (olafleur) [inline client/src/test/java/com/arcbees/beestore/common/dto/ProductDtoTest.java]: nl eof

File: common/src/main/java/com/arcbees/beestore/common/api/ProductResource.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+public interface ProductResource {
+    @GET
+    @Path(ApiPaths.ID)
+    ProductDto getProduct(@PathParam(ApiParameters.ID) int productId,
+                          @QueryParam(ApiParameters.BRAND) String brandValue,
+                          @QueryParam(ApiParameters.SIZE) String size);
+
+    @GET
+    List<ProductDto> getProductsByBrand(@QueryParam(ApiParameters.BRAND) String brand);
+}
```
Comment (olafleur) [inline common/src/main/java/com/arcbees/beestore/common/api/ProductResource.java]: nl eof

**Comment by olafleur** [comment]
A few nits then LGTM.

File: client/src/test/java/com/arcbees/beestore/client/application/product/ProductPresenterTest.java (hunk 3 above)
Comment (Elrhino) [inline client/src/test/java/com/arcbees/beestore/client/application/product/ProductPresenterTest.java:41]: if I take a look at the ArcBees checkstyle, static imports follow the same order as regular imports.

**Comment by Elrhino** [comment]
@olafleur Thanks for the review!

File: client/src/test/java/com/arcbees/beestore/client/application/product/ProductPresenterTest.java (hunk 3 above)
Comment (olafleur) [inline client/src/test/java/com/arcbees/beestore/client/application/product/ProductPresenterTest.java:41]: You are right, I was confused. ;)

**Pull request merged**